---
title: "Publications [(Google Scholar Profile)](https://scholar.google.com/citations?user=vwawLWcAAAAJ&hl=en)"
permalink: /publications/
author_profile: true
---

## Pre-prints
![Yu2018](http://malachycampbell.github.io/images/HpYujpg.jpg){: .align-right height="300px" width="184px"}

<b>[Genomic Bayesian confirmatory factor analysis and Bayesian network to characterize a wide spectrum of rice phenotypes.](http://malachycampbell.github.io/publications/2018-10-6-HP) </b> Yu, H., <b>M.T. Campbell</b>, Q. Zhang, H. Walia, G. Morota. (2018). <i>bioRxiv</i>.

This is a really nice approach to reduce the dimensionality of phenomics datasets and understand the genetic interrelationships between  trait classes. Haipeng used confirmatory factor analysis to reduce 48 observed phenotypes into six latent variables, which essentailly respresent unobserved biological processes that contribute to the traits, and used Bayesian network to understand the interdependence among latent variables. Check out the preprint! 
  
   <i>Currently under review at G3</i>

<br>

![Du2018](http://malachycampbell.github.io/images/Du_2018.jpg){: .align-left height="200px" width="246px"}

<b>[Using LASSO in gene co-expression network for genome-wide identification of gene interactions responding to salt stress in rice.](http://malachycampbell.github.io/publications/2018-6-29-Using-LASSO-in-gene-co-expression) </b> Du, Q., <b>Campbell, M.T.</b>, H. Yu, K. Liu, H. Walia, Q. Zhang, C. Zhang. (2018). <i>bioRxiv</i>. 

Here, we used LASSO to select gene coexpression modules associated with salinity tolerance traits.

<br>

## Peer-reviewed Publications

![Campbell2018RR](http://malachycampbell.github.io/images/Man.gif){: .align-right height="200px" width="436px"}

<b>[Leveraging breeding values obtained from random regression models for genetic inference of longitudinal traits.](http://malachycampbell.github.io/publications/2019-2-9-RRGWAS) </b> <b>M.T. Campbell</b>, M. Momen, H. Walia, G. Morota. (2019). <i>The Plant Genome.</i>.

This study builds on the random regression genomic prediction approach described in [Campbell et al 2018](http://malachycampbell.github.io/publications/2018-05-11-Utilizing-random-regression-models-for-genomic-prediction-of-a-longitudinal-trait-derived-from-high-throughput-phenotyping), and used the derived breeding values for genomic inferenece across time points.

<br>

![Hussain2018](http://malachycampbell.github.io/images/Hussain_2018.jpg){: .align-right height="200px" width="244px"}

<b>[ShinyAIM: Shiny-based Application of Interactive Manhattan Plots for Longitudinal Genome-Wide Association Studies.](http://malachycampbell.github.io/publications/2018-8-1-ShinyAIM) </b> Hussain, W., <b>M.T. Campbell</b>, H. Walia, G. Morota. (2018). <i>Plant Direct (In Press)</i>.

This is a nice tool to easily visualize multiple manhattan plots. Wasseem packaged a bunch of useful methods to explore the phenotypic data and GWAS results into this app. This is particularly useful for phenomics datasets or longitudinal studies.

<br>

![Campbell2018](http://malachycampbell.github.io/images/Campbell_2018_PlantDirect.jpg){: .align-right height="200px" width="191px"}

<b>[Utilizing random regression models for genomic prediction of a longitudinal trait derived from high-throughput phenotyping.](http://malachycampbell.github.io/publications/2018-05-11-Utilizing-random-regression-models-for-genomic-prediction-of-a-longitudinal-trait-derived-from-high-throughput-phenotyping)</b> <b>Campbell, M.T.</b>, H. Walia, and G. Morota. (2018). <i>Plant Direct</i>.

This paper utilized a random regression approach for genomic prediction of longitudinal shoot growth trajectories in rice. The RR approach sigificant imporved prediction accuracies compare to a single time point approach. To our knowledge, this is the first study to implement this approach in a crop species. 

<br>

![Campbell2017_PG](http://malachycampbell.github.io/images/Campbell_2017_PlantGenome.png){: .align-left height="200px" width="186px"}

<b>[A Comprehensive Image based Phenomic Analysis Reveals the Complex Genetic Architecture of Shoot Growth Dynamics in Rice (Oryza sativa).](http://malachycampbell.github.io/publications/2017-06-27-A-Comprehensive-Image-based-Phenomic-Analysis-Reveals-the-Complex-Genetic-Architecture-of-Shoot-Growth-Dynamics-in-Rice)</b> <b>Campbell, M.T. </b>, Q. Du, K. Liu, C.J. Brien, B. Berger, C. Zhang, and H. Walia. (2017). <i>Plant Genome</i>.

We used a "two-step" functional GWAS and genomic prediction approach to study shoot growth dyanamics in rice. First, we modeled shoot biomass over 20 days using a power law function and used model parameters as derived traits for association mapping and genomic prediction.

<br>

![Campbell2017_PlGen](http://malachycampbell.github.io/images/Campbell_2017_PlosGen.png){: .align-right height="250px" width="234px"}

<b>[Allelic variants of OsHKT1;1 underlie the divergence between indica and japonica subspecies of rice (Oryza sativa) for root sodium content.](http://malachycampbell.github.io/publications/2017-06-05-Allelic-variants-of-OsHKT11-underlie-the-divergence-between-indica-and-japonica-subspecies-of-rice-for-root-sodium-content)</b> <b>Campbell, M.T. </b>, N. Bandillo, F.R.A. Al Shiblawi, S. Sharma, K. Liu, Q. Du, A.J. Schmitz, C. Zhang, A.A. Very, A.J. Lorenz, and H. Walia. (2017). <i>PLoS Genetics</i>.

<i>HKT1;1</i> was identified as a major regulator of root sodium content in rice. Variants that influence the transport of sodium were found to be a component of the divergence between to two subspecies of rice for root sodium content.

<br>

![Knecht2016](http://malachycampbell.github.io/images/Knecht_2016.jpeg){: .align-left height="200px" width="243px"}

<b>[Image Harvest: an open source platform for high throughput plant image processing and analysis.](http://malachycampbell.github.io/publications/2016-04-13-Image-Harvest-an-open-source-platform-for-high-throughput-plant-image-processing-and-analysis)</b> Knecht, A.C.&#8224;&#8225;, <b>M.T. Campbell&#8224; </b>, A. Caprez, D.R. Swanson, and H. Walia. (2016). <i>Journal of Experimental Botany</i>. <br>

We developed a software to process and anayze images derived from high throughput phenotyping platforms. <br>
&#8224; Equal contribution <br>
&#8225; Undergraduate coauthor

<br>

![Campbell2015_PlPhys](http://malachycampbell.github.io/images/Campbell_2015_PlantPhys.jpg){: .align-right height="200px" width="331px"}

<b>[Integrating image based phenomics and association analysis to dissect the genetic architecture of temporal salinity responses in rice.](http://malachycampbell.github.io/publications/2015-08-01-Integrating-image-based-phenomics-and-association-analysis-to-dissect-the-genetic-architecture-of-temporal-salinity-responses-in-rice)</b> <b>Campbell, M.T.</b>, A.C. Knecht, B. Berger, C.J. Brien, D. Wang, and H. Walia. (2015). <i>Plant Physiology</i>. <br>

Here, we used image based high thoughput phenotyping and association mapping to examine the genetic basis of temporal salinity responses.

<br>

![Campbell2015_PlosOne](http://malachycampbell.github.io/images/Campbell_2015_PLoSOne.png){: .align-left height="225px" width="233px"}

<b>[Genetic and Molecular Characterization of Submergence Response Identifies Subtol6 as a Major Submergence Tolerance Locus in Maize](http://malachycampbell.github.io/publications/2015-03-25-Genetic-and-Molecular-Characterization-of-Submergence-Response-Identifies-Subtol6-as-a-Major-Submergence-Tolerance-Locus-in-Maize).</b> <b>Campbell, M.T.</b>, Campbell, M.T., C.A. Proctor, Y. Dou, A.J. Schmitz, P. Phansak, G.R. Kruger, C. Zhang, and H. Walia. (2015). <i>PLoS One</i>. <br>

This paper assessed natural variation for submegence tolerance in maize and identifies a QTL that influences submegence tolerance.
 
<br>

![Placido2013](http://malachycampbell.github.io/images/Placido_2013.jpg){: .align-right height="200px" width="341px"}

<b>[Introgression of Novel Traits from a Wild Wheat Relative Improves Drought Adaptation in Wheat](http://malachycampbell.github.io/publications/2013-10-01-Introgression-of-novel-traits-from-a-wild-wheat-relative-improves-drought-adaptation-in-wheat).</b>  Placido, D.F., <b>M.T. Campbell</b>, J.J. Folsom, X. Cui, G.R. Kruger, P.S. Baenziger, and H. Walia. (2013). <i>Plant Physiology</i>. <br>

This paper looks at the effects of an introgression from <i>Agropyron elongatum</i> into <i>Triticum aestivum</i> on drought tolerance.
